nservicebus/azure-service-bus/brokered-message-creation.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Azure Service Bus Transport Brokered Message Creation
+tags:
+- Azure
+- Cloud
+---
+
+This document describes how to modify NServiceBus endpoints in order to set up a native integration between these endpoints and non-NServiceBus endpoints that share an Azure Service Bus namespaces as mutual communication channel.
+
+### BrokeredMessage conventions
+
+#### Brokered Message body format
+
+The Azure Service Bus API allows the construction of a `BrokeredMessage` body from a stream or an object that will get serialized into a byte array by the internals of `BrokeredMessage`.
+
+NOTE: Both the sender (native or NServiceBus) and the receiver must agree on the convention used for sending the message body.
+
+By default the `BrokeredMessage` body is assumed to be a byte array. But for scenarios such as native integration it is usually more appropriate that the body can be retrieved using `Stream` as most non .NET ASB SDK's only support the `Stream` body format. `Stream` does not perform any serialization on the body, so it stays in its raw form on the wire.
```

Comment:
Assumed by the ASB transport to remain compliant with previous versions of NSB. It's better to use `Stream` if possible to avoid the unnecessary serialization of a byte array. Perhaps we should guide towards that.